### ISSUE TITLE
The mount count is completely inconsistent from the Blizzard side.

### ISSUE THREAD

**Issue by seirl** [kevinclement/SimpleArmory#144]
In game, I have 352 mounts on my char according to the statistics tab, but the API only returns 349. SimpleArmory shows I have 335. Why is there such a big disrepancy?

I investigated with the account of Jramm, and added a debug log every time a mount did show in the API result, but wasn't show in the API for side/classes restriction reasons. I got this list:

Wrong faction:

https://wowhead.com/item=159146
https://wowhead.com/item=155662
https://wowhead.com/item=156487
https://wowhead.com/item=161330
https://wowhead.com/item=161773
https://wowhead.com/item=161774
https://wowhead.com/item=161665
https://wowhead.com/item=128526

Wrong class:

https://wowhead.com/spell=5784
https://wowhead.com/item=142231
https://wowhead.com/item=142227
https://wowhead.com/item=143493
https://wowhead.com/item=143492
https://wowhead.com/item=143490
https://wowhead.com/item=143491
https://wowhead.com/item=142225
https://wowhead.com/item=142232
https://wowhead.com/item=143489
https://wowhead.com/item=142228
https://wowhead.com/item=142226
https://wowhead.com/item=142233
https://wowhead.com/item=143637
https://wowhead.com/spell=23161
https://wowhead.com/spell=48778
https://wowhead.com/spell=232412
https://wowhead.com/spell=229377
https://wowhead.com/spell=229376
https://wowhead.com/spell=229417

When I checked those mounts in my in-game journal, it seems that they don't appear for the "wrong faction" ones, but they *do* appear for the "wrong class" ones. Additionally, there's a lot of mounts that do appear on the journal for the wrong class, like the Felsaber, but that aren't returned in the API results.

So basically, in a completely random and inconsistent manner, mounts that aren't usable by your faction or character can or can't count for your mount count. There is no reliable way of knowing if that's the case or not, and there is no known method for accurately reproducing the in-game mount count.

Nothing about that makes any sense for me, so I'd rather wait for Blizzard to clarify their logic or the API, or change their mount counting algorithm to be more consistent.

**Comment by seirl** [comment]
Bug tracked on Blizzard's side here: https://us.battle.net/forums/en/bnet/topic/20770096830#1

**Comment by divinemaiden** [comment]
I'm not sure if this will help but I've found a formula that deals with this as of right now. Don't mean to necro an old issue but new to coding SA :)

In game, I have 484 mounts. SA says 412 mounts (not including the alabaster which I've added locally on my machine so I'm at 413 so add that too). 

So what it doesn't include in SA is the following which you need to add to get in-game Total Mounts (if you have any of the below). I would think this is an API Issue as you listed above but I don't know too much about the API.

- All class mounts acquired (as listed above)
- All opposing faction racial mounts (if you have any of your faction's mounts - you will need to find the corresponding mount)
- All opposing faction Marks of Honor mounts (if you have any of your faction's mounts - you will need to find the corresponding mount)
- All opposing faction Argent Tournament mounts (if you have any of your faction's mounts - you will need to find the corresponding mount) This does not include your main camp faction (Silver Covenant or Sunreavers)
- Nazjatar Questline: Wild Tame/Good Girl - counts both mounts
- Two Sides of Every Tale achievement - counts both mounts
- Felsaber (Demon Hunter)

That's all I know for now so hope that helps!

**Comment by kevinclement** [comment]
I worked on this last week and thought I cracked the code, but the more testing I did I wasn't able to prove it out reliably.  In the end I relaxed restrictions on class mounts but keep faction mounts, but thats not enought to get it to fully line up with in the game.  Feels like we're just at the mercy of blizzard quirky code here and so prob as best as we're gonna get.
